ServiceBus/ServiceBus/ProductService.svc.cs

```diff
@@ -136,63 +141,74 @@ public Result AddProduct(string name, double price, Guid guid)
         /// <param name="guid">ID of a product</param>
         /// <param name="name">New name for a product</param>
         /// <param name="price">New price for a product</param>
+        /// <param name="pType">Type of a product</param>
         /// <returns>Modified product</returns>
-        public SharedLibs.DataContracts.Product EditProduct(Guid guid, string name, double price)
+        public SharedLibs.DataContracts.Product EditProduct(Guid guid, string name, double price, ProductType pType = null, bool enabled = true, bool headliner = false)
         {
             try
             {
-                var originalProduct = GetProduct(guid);
-
-                if (originalProduct.Result.ResultType == ResultType.Success)
-                {
-                    var editableProduct = new Product()
-                    {
-                        Id = originalProduct.ID,
-                        Name = originalProduct.Name,
-                        Price = originalProduct.Price
-                    };
-
                     using (var context = new ServiceBusDatabaseEntities())
                     {
-                        context.Products.Attach(editableProduct);

```

Comment:
I am not sure if this is a correct way how to handle this... I would simply take data from parameters of the method, compare it with object retrieved from context and then update the product. I find this code over complicated.